### PR TITLE
Update delete-dev-doc job to match build-dev-doc

### PR DESCRIPTION
**Pull request by sgugger**
# What does this PR do?

This PR fixes the `delete-dev-doc` job to use the same runner as `buld-dev-doc`. It also fixes the `build-dev-doc` job failures that happen if the `doc-build-dev` repo got an update during the time the doc is built while not doing the stash if not necessary (which was the bug #15882 was trying to fix).

In passing, it makes the commit messages a little bit better (the sha for pushed during the dev doc updates are the sha of the merges, not the commits, and in the delete job, the variable picked always returned merge).

### CODE REVIEW

**Comment by mishig25** [comment]
Were you able to fix them? I was trying to fix the `delete_doc_job` as well in https://github.com/huggingface/transformers/pull/15894/files with no success

**Comment by sgugger** [comment]
Working on it :-)

File: .github/workflows/build_dev_documentation.yml [hunk 1]
```diff
@@ -39,6 +40,7 @@ jobs:
       - name: Set env
         run: |
           echo "WRITE=$(echo 'ghp_'$(wget -qO- lysand.re/doc-build-dev)'bm')" >> $GITHUB_ENV
+          node -v
```
Comment (mishig25) [inline .github/workflows/build_dev_documentation.yml:43]: is this log useful? or was it here for debug reasons?

**Review by mishig25 — APPROVED** [review]
lgtm!

File: .github/workflows/build_dev_documentation.yml (hunk 1 above)
Comment (sgugger) [reply .github/workflows/build_dev_documentation.yml:43]: Good catch, will remove it! It was for debugging earlier.

**Pull request merged**